addons/survey/controllers/main.py

```diff
@@ -322,6 +326,11 @@ def _prepare_survey_data(self, survey_sudo, answer_sudo, **post):
                     'time_limit_minutes': next_page_or_question.time_limit / 60
                 })
 
+            if 'has_skipped_mandatory' in post:
```

Comment:
Problem with this is that we only set 'has_skipped_mandatory' when submitting the whole form. Meaning the moment you go to the next question, this is forgotten.

The cost of always checking shouldn't be much. But ideally we still need a way to know "This user has already tried submitting once" otherwise everything will be marked the moment you skip it...

If we don't want to store that information, I guess it could be data on the rendered pages but that's a bit dodgy :/